### PR TITLE
Add support for external loss function definition

### LINKED ISSUE #983: Add support for external loss functions
It should be possible to use `torch.hub` to load external loss functions the same way that it is used to load external architectures (#985). This will allow the use of arbitrary loss functions, which is currently not possible due to the requirement that the configs be serializable.

@lewfish has suggested that instead of adding focal loss to RV (#982), we can have it in a separate repo and use it through this mechanism where needed.

### PR DESCRIPTION
**Pull request by AdeelH**
## Overview

Adds support for loading the loss function via torch.hub. Currently not supported for object detection.

- An external loss definition can be specified by supplying an `ExternalModuleConfig` to `SolverConfig.external_loss_def`.
- Example usage:
	- classification (`spacenet_rio.py`)
	```python3
    if external_loss:
        external_loss_def = ExternalModuleConfig(
            github_repo='AdeelH/pytorch-multi-class-focal-loss',
            name='focal_loss',
            entrypoint='focal_loss',
            force_reload=False,
            entrypoint_kwargs={
                'alpha': [.75, .25],
                'gamma': 2
            })
    else:
        external_loss_def = None

    solver = SolverConfig(
        lr=1e-2,
        num_epochs=20,
        test_num_epochs=4,
        batch_sz=32,
        one_cycle=True,
        external_loss_def=external_loss_def)
	}))
	```
- Example usage included in `spacenet_rio.py`, and can be enabled through the `external_loss` argument.
- When loading from a bundle, the definition is recovered and its path provided to `learner.__init__()` via a new `loss_def_path` param.
- Everything else works the same way as in #985.

### Checklist

- [x] Updated `docs/changelog.rst`
- [x] Added `needs-backport` label if PR is bug fix that applies to previous minor release
- [x] Ran scripts/format_code and committed any changes
- [x] Documentation updated if needed
- [x] PR has a name that won't get you publicly shamed for vagueness

## Testing Instructions

* How to test this PR
    - Use example above. Try with different options.

Closes #983

### CODE REVIEW

File: rastervision_pytorch_backend/rastervision/pytorch_backend/examples/chip_classification/spacenet_rio.py [hunk 1]
```diff
@@ -96,8 +97,27 @@ def make_scene(scene_info):
...
+            entrypoint_kwargs={
+                'alpha': [.75, .25],
+                'gamma': 2
+            })
+    else:
+        external_loss_def = None
+
     solver = SolverConfig(
-        lr=1e-4, num_epochs=20, test_num_epochs=3, batch_sz=32, one_cycle=True)
+        lr=1e-2,
```
Comment (lewfish) [inline rastervision_pytorch_backend/rastervision/pytorch_backend/examples/chip_classification/spacenet_rio.py:114]: Did you mean to change the LR? I would have thought that LR would not work.

File: rastervision_pytorch_learner/rastervision/pytorch_learner/learner.py [hunk 2]
```diff
@@ -211,37 +215,75 @@ def setup_model(self, model_def_path: str = None) -> None:
...
         pass
 
+    def setup_loss(self, loss_def_path: Optional[str] = None) -> None:
+        """Setup self.loss.
+
+        Args:
+            loss_def_path (str, optional): Loss definition path. Will be
+            available when loading from a bundle. Defaults to None.
+        """
+        extCfg = self.cfg.solver.external_loss_def
```
Comment (lewfish) [inline rastervision_pytorch_learner/rastervision/pytorch_learner/learner.py:238]: Is there a reason why `extCfg` isn't named `ext_cfg` (following the usual Python naming conventions)?

**Review by lewfish — CHANGES_REQUESTED** [review]

File: rastervision_pytorch_backend/rastervision/pytorch_backend/examples/chip_classification/spacenet_rio.py (hunk 1 above)
Comment (AdeelH) [reply rastervision_pytorch_backend/rastervision/pytorch_backend/examples/chip_classification/spacenet_rio.py:114]: Nope. Didn't mean to commit this.

File: rastervision_pytorch_learner/rastervision/pytorch_learner/learner.py (hunk 2 above)
Comment (AdeelH) [reply rastervision_pytorch_learner/rastervision/pytorch_learner/learner.py:238]: That's a good point. I have no idea why I didn't notice that before.

**Review by lewfish — APPROVED** [review]
